torch/optim/optimizer.py

```diff
@@ -113,15 +116,17 @@ def _stack_if_compiling(x):
         return x
 
 
-def _disable_dynamo_if_unsupported(single_tensor_fn=None):
+def _disable_dynamo_if_unsupported(
+    single_tensor_fn: Optional[Callable[..., object]] = None
```

Comment:
```suggestion
    single_tensor_fn: Optional[Callable[_P], _T]] = None
```
or is there some reason that doesn't work?

Reply:
It's not correct, is it? The single_tensor_fn isn't actually related to the returned value in any way (at least not directly in a way the decorator can see). For example look at the use of this on `adamax`. The `single_tensor_fn` has less varargs and more kwargs than the decorated function.

Reply:
Ah good catch, could be its own paramsspec object though if the args/kwargs are useful